src/pick.c

```diff
@@ -446,6 +448,14 @@ selected_choice(void)
 			while (cursor_position < query_length
 			    && isu8cont(query[++cursor_position]));
 			break;
+		case PAGE_DOWN:
+			if (yscroll + lines - 1 < choices_count)
```

Comment:
Great input! I agree, the current behaviour is confusing. I followed
your advice and made the page down/up move the selection one page
down/up, counting from the current selection. If less choices than a
full page is available the first/last choice is selected. The man-page
is updated, do you think it describes the behaviour clear enough?